lib/ddtrace/contrib/sidekiq/server_tracer.rb

```diff
@@ -16,7 +16,8 @@ def initialize(options = {})
         def call(worker, job, queue)
           resource = job_resource(job)
 
-          service = service_from_worker_config(resource) || @sidekiq_service
+          service = worker_config(resource, :service_name) || @sidekiq_service
+          log_args = worker_config(resource, :log_job_arguments) || configuration[:log_job_arguments]
```

Comment:
This is the best I could come up with for testing with the flag. Let me know if there is a better way.